src/libstore/globals.hh

```diff
@@ -142,6 +142,30 @@ public:
         )",
         {"build-cores"}, false};
 
+    Setting<unsigned int> loadLimit{
+      this,
+      getDefaultCores(),
+      "load-limit",
+      R"(
+        Sets the value of the `NIX_LOAD_LIMIT` environment variable in the
+        invocation of builders. Builders can use this value at their discretion
+        to dynamically control the amount of parallelism with respect to the
+        machine's load average. For instance, in Nixpkgs, if this value is >0,
+        the builder passes the `-lN` flag to GNU Make. In this case, if the
+        load average of the machine exceeds `N`, the amount of parallelism will
+        be dynamically reduced to 1.
+
+        By default, it is set to the number of cores available to Nix.
+
+        On busy machines where Nix co-exists with other workloads, parallelism
+        may not work as intended. For example, consider a 64 core machine whose
+        load average is 24 and where Nix is limited to 8 cores. By default,
+        `-j8 -l8` will be passed to GNU Make. Since the load average exceeds 8,
```

Comment:
```suggestion
        load average is 24 and where Nix is limited to 8 cores. By default,
        `-j8 -l8` would be passed to GNU Make. Since the load average exceeds 8,
```

Since we're still in the example where the builder is a Make invocation.